Feature/wpebrowser 54

BunioFH opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/rdkcentral/meta-dac-sdk?pullRequest=38) <br/>All committers have signed the CLA.

BunioFH committed:
WPEBROWSER-54 cherry picked lifecycle 2.0 into wpe-webkit

change https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/3035cf746fbbc8ce60d19158806a8d1006bed95e
from https://github.com/WebPlatformForEmbedded/WPEWebKit/tree/wpe-2.38_page_lifecycle_api_new
layed on top of 2.38.8

BunioFH committed:
WPEBROWSER-54 minimize wpeframework inside DAC

BunioFH committed:
WPEBROWSER-54 firebolt-native-transport

BunioFH committed:
WPEBROWSER-54 firebolt-api-cpp

BunioFH committed:
WPEBROWSER-54 cherry picked lifecycle 2.0 in cog

change https://github.com/pgorszkowski-igalia/cog/commit/6aa8dc6d5663b326495c7d30669c578066feeb00
from https://github.com/pgorszkowski-igalia/cog/tree/0.16.0_page_lifecycle_new
layed on top of cog 0.16.1

BunioFH committed:
WPEBROWSER-54 added system dbus to cog

BunioFH committed:
WPEBROWSER-54 basic firebolt platform in cog

supports:
* platform registration
* lifecycle init on new WebKitWebView
* signals from WebKitWebView
	- here's where W3C lifecycle events should come from
	  most of that is either missing or unclear in WPE api
* Firebolt connectivity & app events registration
	- here's where COG lifecycle calls should happen
* multiple WebViews are not supported
  don't seem to be properly handled in COG as well
